Week 11 - Project Happy Thoughts - Julia Ros

julialeonieros opened the pull request:
Link to my deployed project:
https://determined-engelbart-a7f7f6.netlify.app/

My reflections on how this week's project turned out:
I aimed for the app to resemble the image in the project brief, and I am quite happy with the result. If I would have had more time I'd probably add an animation. For now I only added a character counter.

Things I'd like to have clarified or explained in more detail:
I just realised I forgot to make the site responsive, and I'll try to do that in the near future.

Ann-Sofi commented on code/src/App.js:
```diff
@@ -1,9 +1,77 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+
```
Good that you left a empty line between the outer and local imports.

Ann-Sofi commented on code/src/App.js:
```diff
@@ -1,9 +1,77 @@
-      Find me in src/app.js!
+    <div className="main-container">
+      <Form 
+        onSubmit={handleFormSubmit}
+        value={messageNew}
+        onChange={handleMessageNewChange}
```
Good naming with the "handle". Now I know that this is beeing forwarded as props to a component.

Ann-Sofi commented on code/src/Components/Form.js:
```diff
@@ -0,0 +1,36 @@
+          </form>
+        </div>
+        <div>
+          <p className={value.length > 134 ? "character-counter red" : "character-counter black" }>Characters left: {(140-value.length)} 📝</p>
```
Nice and short solution! :-)

Ann-Sofi reviewed (approved):
Cute site and it works as it should at the blue level. 

Good commit messages. I can follow your process by looking at them 👍

SInce you added a "warning" when the text is getting to long, I think the next thing to add could be to add a warning when it's to short. It could be pretty easy to reach the maximum of letters, but it might be just as common to write a very short one?

KarolCislowski commented on code/src/App.js:
```diff
@@ -1,9 +1,77 @@
+  }
+
+  const refreshPage = () => {
+    window.location.reload()
```
It is an anti-pattern to reload single page application. You need to rethink the state/props structure to make your app re-render

KarolCislowski commented on code/src/App.js:
```diff
@@ -1,9 +1,77 @@
+    fetch(LIKES_URL(id), options)
+      .then(result => result.json())
+      .then(() => fetchMessageList())
+      .catch(error => console.error(error))
```
console.log can't be present in the production code.

KarolCislowski reviewed (changes requested):
Hej!
Great job, but please:
1. clean your code from all console.log. 🧹 
2. find a way to avoid refreshing.
3. Make it responsive

julialeonieros commented:
Hi Karol!
I've now made it responsive and removed the refresh-function (instead set the state of messageNew to empty string). I also removed the console.errors (although Maks used console.errors like that in the lecture so I thought it was ok, but I guess you should remove them when you see that the code works?).

KarolCislowski reviewed (approved):
Awesome app!!
